internal/bffauth/client.go

```diff
@@ -0,0 +1,217 @@
+package bffauth
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/bluesky-social/indigo/atproto/crypto"
+	"github.com/eagraf/habitat-new/internal/resolvers"
+	"github.com/pkg/errors"
+)
+
+type ExternalHabitatUser struct {
+	DID       string
+	PublicKey crypto.PublicKey
+	Host      string
+}
+
+func getTempStores() (map[string]crypto.PrivateKey, map[string]*ExternalHabitatUser, error) {
+	testKey, err := crypto.GeneratePrivateKeyP256()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	testPubKey, err := testKey.PublicKey()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	testUser := &ExternalHabitatUser{
+		DID:       "did:plc:avkzrsj7u4kmoq33gx6v4lh4",
+		PublicKey: testPubKey,
+		Host:      "beacon-dev.tail07d32.ts.net",
+	}
+
+	return map[string]crypto.PrivateKey{
+			"did:plc:avkzrsj7u4kmoq33gx6v4lh4": testKey,
+		}, map[string]*ExternalHabitatUser{
+			"did:plc:avkzrsj7u4kmoq33gx6v4lh4": testUser,
+		}, nil
+
+}
+
+type Client interface {
+	GetToken(targetDID string) (string, error)
+}
+
+type client struct {
+	clientDID string
+
+	privateKey crypto.PrivateKey
+
+	activeTokens map[string]string
+
+	// HTTP or HTTPS
+	scheme string
```

Comment:
hmmm why do we need this? can we just decide on one? (need https for security right)

Reply:
I ran into it because of the httptest server in the unit tests